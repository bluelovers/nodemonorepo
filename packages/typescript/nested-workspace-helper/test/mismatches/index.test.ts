import getMismatchedDependencies, {Checker} from '../../lib/mismatches'
import createSetupTeardown from '../lib/setup-teardown'
import allCheckers from '../lib/all-mismatch-checkers'

const {apply} = createSetupTeardown('mismatched-deps.yaml')
const root = 'root'

function iterateCheckers (fn: (check: Checker, name: string) => void) {
  Object
    .entries(allCheckers)
    .forEach(([name, check]) => fn(check, name))
}

describe('getMismatchedDependencies', () => {
  iterateCheckers((check, checkerName) => {
    describe(`when checker is ${checkerName}`, () => {
      const get = () => getMismatchedDependencies(root, check)
      it('matches snapshot', apply(async () => {
        expect(await get()).toMatchSnapshot()
      }))
    })
  })
})
