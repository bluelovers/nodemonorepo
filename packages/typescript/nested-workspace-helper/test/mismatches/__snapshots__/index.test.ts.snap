// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMismatchedDependencies when checker is ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_CARET_EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_CARET_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_EQUAL_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_TILDA_EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is ANY_OR_TILDA_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "*",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "*",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is CARET_EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "^1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "^3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is CARET_EQUAL_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "^1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "^3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is CARET_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "^3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is CARET_MIN_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "^3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is EQUAL_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is EQUAL_MIN_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is EQUAL_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is TILDA_EQUAL matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "~1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "~3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is TILDA_EQUAL_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "~1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "~3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is TILDA_MIN matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "~3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;

exports[`getMismatchedDependencies when checker is TILDA_MIN_OR_ANY matches snapshot 1`] = `
Object {
  "root": Array [
    Object {
      "name": "matched000",
      "requirement": "1.2.3",
      "type": "prod",
      "update": "1.2.3",
      "version": "1.2.3",
    },
    Object {
      "name": "mismatched000",
      "requirement": "^2.2.3",
      "type": "prod",
      "update": "~3.2.1",
      "version": "3.2.1",
    },
  ],
  "root/matched/matched000": Array [],
  "root/mismatched/mismatched000": Array [],
}
`;
